src/core/store/modules/vaults/vaults.reducer.ts

```diff
@@ -279,7 +279,7 @@ const vaultsReducer = createReducer(vaultsInitialState, (builder) => {
       state.statusMap.getExpectedTransactionOutcome = {};
     })
     .addCase(getExpectedTransactionOutcome.rejected, (state, { error }) => {
-      state.statusMap.getExpectedTransactionOutcome = { error: 'Simulation Failed' };
+      state.statusMap.getExpectedTransactionOutcome = { error: error.message };
```

Comment:
wdyt if we use `parseError(error)` here in the reducer on the `rejected` actions, instead of wrapping the thrown error on the thunks?

This way we would have the whole thunk covered and no need for using try catch.

Reply:
++
I was about to comment the same exact thing

Reply:
++

Reply:
I think we have two options:
1) We adapt sdk error structure to the SerializedError structure (Seems only error_code would be required to change)
```
export interface SerializedError {
  name?: string // error_type => name ?
  message?: string
  stack?: string
  code?: string  // error_code => code
}
```
2) We add a custom `serializeError()` function to each thunk that needs it: https://redux-toolkit.js.org/api/createAsyncThunk#options

Reply:
Intention is not to couple it to a dependency, but making current naming convention more close to a standard, since the SerializedError interface is somewhat more standard than the one the sdk is currently using. Im not suggesting replacing the one we have on sdk with this one, just adapting the names basically. Think that change shouldnt avoid us from having the typescript benefits. Sdk errors can still have more properties as needed.

About wrapping the `createAsyncThunk` function, yes it can be done, and can be used accordingly depending on the use case. Its just the way the library allow us.

Im more inclined to option 1 though. Lets see what @gotzenx and @turtlemoji think.

Reply:
Will approve since the custom serializeError is a good solution imo. We can see later if we decide to refactor :)